pkg/plugin/processor/builtin/avro/decode.go

```diff
@@ -0,0 +1,206 @@
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/conduitio/conduit/pkg/plugin/processor/builtin/avro/schemaregistry"
+	"github.com/lovromazgon/franz-go/pkg/sr"
+)
+
+type decoder interface {
+	Decode(ctx context.Context, b opencdc.RawData) (opencdc.StructuredData, error)
+}
+
+type decodeConfig struct {
+	// The field that will be encoded.
+	Field string `json:"field" default:".Payload.After"`
+
+	// URL of the schema registry (e.g. http://localhost:8085)
+	URL string `json:"url" validate:"required"`
+
+	Auth authConfig `json:"auth"`
+	TLS  tlsConfig  `json:"tls"`
+
+	fieldResolver sdk.ReferenceResolver
+}
+
+func parseDecodeConfig(ctx context.Context, m map[string]string) (decodeConfig, error) {
+	cfg := decodeConfig{}
+	err := sdk.ParseConfig(ctx, m, &cfg, cfg.Parameters())
+	if err != nil {
+		return decodeConfig{}, err
+	}
+
+	err = cfg.Auth.validate()
+	if err != nil {
+		return decodeConfig{}, cerrors.Errorf("invalid basic auth: %w", err)
+	}
+
+	err = cfg.TLS.parse()
+	if err != nil {
+		return decodeConfig{}, cerrors.Errorf("failed parsing TLS: %w", err)
+	}
+
+	// Parse target field
+	rr, err := sdk.NewReferenceResolver(cfg.Field)
+	if err != nil {
+		return decodeConfig{}, cerrors.Errorf("failed parsing target field: %w", err)
+	}
+	cfg.fieldResolver = rr
+
+	return cfg, nil
+}
+
+func (c decodeConfig) ClientOptions() []sr.Opt {
+	clientOpts := []sr.Opt{sr.URLs(c.URL), sr.Normalize()}
+	if c.Auth.Username != "" && c.Auth.Password != "" {
+		clientOpts = append(clientOpts, sr.BasicAuth(c.Auth.Username, c.Auth.Password))
+	}
+
+	if c.TLS.tlsClientCert != nil {
+		tlsCfg := &tls.Config{
+			Certificates: []tls.Certificate{*c.TLS.tlsClientCert},
+			MinVersion:   tls.VersionTLS12,
+		}
+		if c.TLS.tlsCACert != nil {
+			tlsCfg.RootCAs = c.TLS.tlsCACert
+		}
+		clientOpts = append(clientOpts, sr.DialTLSConfig(tlsCfg))
+	}
+
+	return clientOpts
+}
+
+type decodeProcessor struct {
+	sdk.UnimplementedProcessor
+
+	logger  log.CtxLogger
+	cfg     decodeConfig
+	decoder decoder
+}
+
+func NewDecodeProcessor(logger log.CtxLogger) sdk.Processor {
+	return &decodeProcessor{logger: logger}
+}
+
+func (p *decodeProcessor) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:    "avro.encode",
+		Summary: "Decodes a field's raw data in the Avro format",
+		Description: `The processor takes raw data (bytes or a Base64-encoded string) in the specified field and decodes 
+it from the [Avro format](https://avro.apache.org/) into structured data. It extracts the schema ID from the data, 
+downloads the associated schema from the [schema registry](https://docs.confluent.io/platform/current/schema-registry/index.html)
+and decodes the payload. The schema is cached locally after it's first downloaded. 
+
+Currently, the processor only supports the Avro format. If the processor encounters structured data or the data 
```

Comment:
Well yeah, it's the `avro.decode` processor 😄 This is probably taken from the old processor, which was a "generic" processor.
```suggestion
If the processor encounters structured data or the data 
```

Reply:
Ah, exactly!